delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/indexing/IndexingStreamBehaviour.scala

```diff
@@ -0,0 +1,284 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.indexing
+
+import akka.actor.typed._
+import akka.actor.typed.scaladsl.Behaviors
+import akka.cluster.sharding.typed.scaladsl.ClusterSharding
+import akka.cluster.sharding.typed.scaladsl.ClusterSharding.ShardCommand
+import ch.epfl.bluebrain.nexus.delta.kernel.RetryStrategy
+import ch.epfl.bluebrain.nexus.delta.kernel.utils.UUIDF
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.ProjectRef
+import ch.epfl.bluebrain.nexus.delta.sdk.model.views.ViewIndex
+import ch.epfl.bluebrain.nexus.delta.sourcing.projections.stream.StreamSwitch
+import ch.epfl.bluebrain.nexus.delta.sourcing.projections.{Projection, ProjectionProgress}
+import com.typesafe.scalalogging.Logger
+import fs2.Stream
+import monix.bio.{Task, UIO}
+import monix.execution.Scheduler
+
+import java.util.UUID
+import scala.util.{Failure, Success}
+
+/**
+  * The view index behavior defined in order to manage the indexing stream
```

Comment:
It took me a while to follow this. An extra documentation with the different behaviour and under which conditions you jump from a behaviour to the next would be useful